instrumentation/ruby_kafka/lib/opentelemetry/instrumentation/ruby_kafka/utils.rb

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Instrumentation
+    module RubyKafka
+      # Utilities to help with instrumenting kafka
+      module Utils
+        module_function
+
+        def encode_message_key(key)
+          return key if key.encoding == Encoding::UTF_8 && key.valid_encoding?
```

Comment:
This is quite interesting to me. Does Kafka support binary message keys?

It also looks similar to an existing utility method [`OpenTelemetry::Common::Utilities::utf8_encode`](https://github.com/open-telemetry/opentelemetry-ruby/blob/2bf0e6aed060867bc65560a92e2c464d48b26a02/common/lib/opentelemetry/common/utilities.rb#L47) however I see that this implementation interprets the binary as a HEX string. 

I feel like this method should either use or enhance the exiting utilities method.

Reply:
Yes, Kafka will accept any collection of bytes as the message key. I have seen:
- UUIDs
- DB row keys
- Integer values

When investigating the error, we found a client producing messages where the message key is a Java Long (derived from DB row key) and this was causing the Otel exporters to errors.

Yes, the method you've shared does look similar. I was unsure about using as it would cause my example to display a message key of `''` as all characters would be replaced. However, I would like to get this fixed so I'm happy to go with whatever the prior art is.

Reply:
Thanks for the explanation. 

Is this how other dynamic language instrumentations handle this? 

Would it make sense to always the key as if where binary of the predicate fails in stress of forcing the encoding to raise an exception?

Reply:
Oofff my last comment was auto completed into nonsense.

I edited it 🤦

Reply:
For this case, I've checked against other Kafka instrumentation implementations and none I've checked so far setting this attribute so unfortunately there's nothing to compare with.

However, I just had another look at the [specification](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/messaging.md#apache-kafka) and I've noticed there's a footer for the message key:
> If the key type is not string, it's string representation has to be supplied for the attribute. If the key has no unambiguous, canonical string form, don't include its value.

My reaction to this then is that we should not include keys that can't be encoded rather than try to encode or represent the data in another form. Would you agree then that the correct behaviour should be to not set the attribute if utf8 encoding produces a string with invalid encoding?